haskellPackages.haskell-language-server: 0.8.0 -> 0.9.0

maralorn opened the pull request:
Here we go.

The version bump happened by hackage, I just needed to make it build successfully via overrides.

\cc @Lucus16 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

maralorn committed:
haskellPackages.haskell-language-server: 0.8.0 -> 0.9.0

maralorn commented:
@GrahamcOfBorg build haskell-language-server

cdepillabout commented on pkgs/development/haskell-modules/configuration-common.nix:
```diff
@@ -1410,10 +1410,14 @@ self: super: {
+  hls-tactics-plugin = doJailbreak super.hls-tactics-plugin;
+
+  # 2021-02-08: Overrides because nightly is to old for hls 0.9.0
+  lsp-test = doDistribute (dontCheck self.lsp-test_0_11_0_7);
+  haskell-lsp = doDistribute self.haskell-lsp_0_23_0_0;
+  haskell-lsp-types = doDistribute self.haskell-lsp-types_0_23_0_0;
```
I'm surprised these get marked as `hydraPlatforms = none;`.  Do you have any idea why this would be happening?

maralorn replied:
`extra-packages` always do. We could have a debate about this, but I think it makes sense, at least when we consider that this rule applies for every extra package version we get when something is newer on hackage than on stackage.

cdepillabout replied:
Oh, huh, I didn't realize this!

maralorn committed:
Re-enable Cabal 3.2.* build for older ghc versions

maralorn commented:
@GrahamcOfBorg build haskell-language-server

maralorn commented:
@peti The Cabal 3.4 release breaks a lot of overrides. I have a commit in here to fix that. (Because it affects the hls builds for older versions.) Besides the question if this belongs in this PR, do you think pinning 3.2.1 this way is the right solution? There are tens of overrides in the configuration-ghc-*.nix files for older ghc versions where Cabal_3_2_1_0 is explicitely mentioned. I have no clue with which of these Cabal_3_4_0_0 would be compatible.

maralorn commented:
I am a bit weary, that we also should merge this today. Last weeks haskell-updates merge took quite a while because of a late hls merge, wouldn‘t want to repeat that.

maralorn committed:
haskellPackages.haskell-language-server: Pin dependencies

While the dependencies are being released for hls 0.10.0 they break
compatibility with hls 0.9.0 and therefor all need manual intervention
when they get released until hls 0.10.0 get’s released.

maralorn commented:
I am going ahead in merging this because I want to hear hydras opinion about it.

If the Cabal pinning is a bad idea we can still change it tomorrow.

teto commented:
@maralorn thanks for your diligent work on updating hls. Do you have more insight as to when/how each release gets uploaded to hackage ? https://hackage.haskell.org/package/haskell-language-server I am especially interestd about 1.0 since it delivers new sexy features.

cdepillabout commented:
@teto It looks like there is an upstream issue about it: https://github.com/haskell/haskell-language-server/issues/1470

maralorn commented:
@teto I generally lurk in the hls freenode channel where they coordinate the development. The hackage release is a big pain point because of the project structure. Last time they did it in about 4 days, but I assume without better automation motivation to churn trough that will decrease with every release.

But yeah, I have been contacted multiple times about 1.0 and I am really exited about it, too!
